compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/extended/UnusedChecker.kt

```diff
@@ -35,42 +38,53 @@ object UnusedChecker : FirControlFlowChecker() {
     }
 
     class CfaVisitor(
-        val data: Map<CFGNode<*>, VariableStatusInfo>,
+        val data: Map<CFGNode<*>, PathAwareVariableStatusInfo>,
         val reporter: DiagnosticReporter
     ) : ControlFlowGraphVisitorVoid() {
         override fun visitNode(node: CFGNode<*>) {}
 
         override fun visitVariableAssignmentNode(node: VariableAssignmentNode) {
             val variableSymbol = (node.fir.calleeReference as? FirResolvedNamedReference)?.resolvedSymbol ?: return
-            val data = data[node]?.get(variableSymbol) ?: return
-            if (data == VariableStatus.ONLY_WRITTEN_NEVER_READ) {
-                // todo: report case like "a += 1" where `a` `doesn't writes` different way (special for Idea)
-                val source = node.fir.lValue.source
-                reporter.report(source, FirErrors.ASSIGNED_VALUE_IS_NEVER_READ)
+            val dataPerNode = data[node] ?: return
+            for (label in dataPerNode.keys) {
+                val data = dataPerNode[label]!![variableSymbol] ?: continue
```

Comment:
May be it's better to enumerate entries `for ((label, pathInfo) in dataPerNode)` and remove this `!!`?

Reply:
Ah, right. I don't know why I've stuck to use `label`. There could be many different per-label iterations. Thanks to these comments, I revisited all my previous iterations of path-aware infos and remove `!!`.